shell: Add a configuration for not printing shell errors

Brianmm94 opened the pull request:
The shell currently prints out `: command not found` and
`Please specify a subcommand.` when invalid commands are
sent.

This can cause issues in certain situations, such as if a U-Boot
console in interactive mode is on the other end of the shell's
UART, where the U-Boot console will stop booting the device
if it receives any characters, and it will print out strings that
the shell then sees as commands. This causes the shell to send
out the messages above, preventing the device running
U-Boot on the other end from booting up.

I added the configurations
`CONFIG_SHELL_MSG_SPECIFY_SUBCOMMAND` and
`CONFIG_SHELL_MSG_CMD_NOT_FOUND` to allow disabling
these messages.

Signed-off-by: Brian Moran <brian.moran@sabantoag.com>

jakub-uC commented:
Depending on the context, might turning off shell echo address this issue?
I'm wary that the approach you've suggested might be overly drastic. If you could share more details about the specific scenarios where shell message output is a concern, perhaps we could identify a more refined solution. If possible, I would like to explore less invasive alternatives

Brianmm94 commented:
@jakub-uC Hello, setting `SHELL_ECHO_STATUS=n` and `CONFIG_SHELL_PROMPT_UART=""` partially address the issue by no longer printing the echo and prompt, but the shell will still print out the `SHELL_ERROR` messages I mentioned above when it receives messages from U-Boot, such as `Loading` or `>`. If preferred, I could move the configuration checks to `shell.c` lines 531, 695, and 742 seeing as printing `SHELL_MSG_CMD_NOT_FOUND` and `SHELL_MSG_SPECIFY_SUBCOMMAND` are the messages that cause the issue. I could also create a separate configuration for each of them if you think it'd be beneficial.

jakub-uC commented:
@Brianmm94 
In my opinion, your new suggestion is very good and aligns more closely with our current implementation, for example, the shell's CONFIG_SHELL_HELP printing functionality. I think it will be more coherent and potentially less disruptive.
Thank you for the alternative!

Brianmm94 committed:
shell: Add a configuration for not printing shell messages

The shell currently prints out `: command not found` and
`Please specify a subcommand.` when invalid commands are
sent.

This can cause issues in certain situations, such as if a U-Boot
console in interactive mode is on the other end of the shell's
UART, where the U-Boot console will stop booting the device
if it receives any characters, and it will print out strings that
the shell then sees as commands. This causes the shell to send
out the messages above, preventing the device running
U-Boot on the other end from booting up.

I added the configurations
`CONFIG_SHELL_MSG_SPECIFY_SUBCOMMAND` and
`CONFIG_SHELL_MSG_CMD_NOT_FOUND` to allow disabling
these messages.

Signed-off-by: Brian Moran <brian.moran@sabantoag.com>

Brianmm94 commented:
Updated! I used the names `CONFIG_SHELL_MSG_CMD_NOT_FOUND` and `CONFIG_SHELL_MSG_SPECIFY_SUBCOMMAND` to match the existing `SHELL_MSG_CMD_NOT_FOUND` and `SHELL_MSG_SPECIFY_SUBCOMMAND` #defines.

jakub-uC commented:
Hi @Brianmm94, I had another thought that I wanted to share with you. Would you mind checking if the `CONFIG_SHELL_AUTOSTART=n` config meets your needs for your use case? You can call `shell_start` after booting up. Just a suggestion, and I'd love to hear your thoughts on this!

Brianmm94 commented:
Hello @jakub-uC, that's actually what we started doing at first, but that solution only works if the U-Boot device on the other end just boots up once when the product is first powered on. If the U-Boot device power cycles for any reason after that, then the issue still occurs. For example, in production we power off the U-Boot device to reduce power draw when it's not needed and we have APIs for power cycling it.

henrikbrixandersen commented:
Just out of curiosity, why is the Zephyr console and the U-boot console connected together in the first place?
